### ISSUE TITLE
Checklist: getting JULES.jl ready to merge with Oceananigans.jl

### ISSUE THREAD

**Issue by ali-ramadhan** [thabbott/JULES.jl#70]
X-Ref: https://github.com/CliMA/Oceananigans.jl/issues/605

Opening this issue to keep track of what needs to be done before we think JULES.jl and its `CompressibleModel` can be merged into Oceananigans.jl.

We just released Oceananigans v0.38.0 yesterday whiles JULES.jl is relying on Oceananigans v0.24.1 so I'll do some updating first then see what needs to be done.

Probably want JULES to be running on the GPU before merging.

**Comment by ali-ramadhan** [comment]
Hmmm, actually not sure whether merging the two packages is the best approach... But might as well work some more on JULES.jl and worry about whether to merge or not later.

**Comment by ali-ramadhan** [comment]
Hmmm, maybe this issue can be a checklist of milestones before we have something interesting enough to consider submitting a JOSS paper?

Abstractions for base states (#50) and maybe a first microphysics scheme might be nice since it's developed for atmospheric applications, but we can think about that once the current list of milestones are exhausted.
